src/createAsset.ts

```diff
@@ -108,32 +148,23 @@ export class CreateAsset {
 			lte: `${DB_KEY_BLOCKS_HEIGHT}:${formatInt(snapshotHeight)}`,
 		});
 
-		const blocksHeader = await getDataFromDBStream(blocksStream, blockHeaderSchema);
-		const blocks: Block[] = await Promise.all(
-			blocksHeader.map(async (blockHeader: string | Buffer) => {
-				const blockID = hash(blockHeader);
-				const encodedTransactions = await getTransactions(blockID, this._db);
-				const decodedTransactions = await encodedTransactions.map(async tx => {
-					const transactions = await codec.decode<{
-						[key: string]: unknown;
-						asset: Buffer;
-						moduleID: number;
-						assetID: number;
-					}>(transactionSchema, tx);
-
-					const id = hash(tx);
-					return {
-						...transactions,
-						id,
-					};
-				});
-
-				return {
-					blockHeader,
-					payload: decodedTransactions,
-				};
-			}),
-		);
+		const arrayOfBlockIds = await getBlocksIDsFromDBStream(blocksStream);
+		const blocksHeader = await getBlockHeadersByIDs(this._db, arrayOfBlockIds);
```

Comment:
Please rename `arrayOfBlockIds` to `blockIDs` wherever applicable.
```suggestion
		const blockIDs: Buffer[] = await getBlocksIDsFromDBStream(blocksStream);
		const blocksHeader = await getBlockHeadersByIDs(this._db, blockIDs);
```